source/tinyxml2.cpp

```diff
@@ -19,6 +19,10 @@ must not be misrepresented as being the original software.
 
 3. This notice may not be removed or altered from any source
 distribution.
+
+
+Modifications:
```

Comment:
The license explicitly states that modifications are allowed, as long as these modifications are explicitly marked as that (what I did), and the license itself stays in place unmodified (what it is). 

The modification is a single line, which can in most cases be copied again if the source is updated. And even if the modification does not get copied over to a newer version (because it no longer works etc.) , it does not break anything (the only effect is that `&apos;` will appear again in the XML files). I made the modification with exactly these points in mind.

Reply:
Right, I get that it's allowed, It was more that modifying the library directly is kind of error-prone from a maintainability standpoint, as no one is actually going to check the tinyxml source files for modifications before updating it, since they'd reasonably expect the behavior to be the same across new versions (especially any new contributors to the project).

Is it not feasible with the interface to disable entity processing, so that this isn't necessary internally in the library?

Reply:
> Is it not feasible with the interface to disable entity processing, so that this isn't necessary internally in the library?
No, we can not disable the entity processing completely (what would be possible through the interface), because then the XML files will break as soon as there is a location or setting with an `&` in its name. (This is currently not the case, but is thinkable to happen at some point.) Disabling only the `'` is not possible through the interface.

If the change in the library is not wanted, I can remove it, so that `'` will stay as `&apos;` in the XML files.

Reply:
The location list will probably not be changing, besides probably adding Medigoron and Carpet salesman at one point, unless someone decides to implement rupeesanity or something, even then I'm not sure what locations would have & in their name. I looked through a text dump of the game and & is nowhere to be found except the credits, so I don't think it would come up.

Settings I can't be as sure of though, unless we intentionally avoid it. I mean even the aforementioned Medigoron/Carpet Salesman option could be named something like "Shuffle Medigoron & Carpet Salesman". Personally I would be fine with just avoiding & in setting names in order to make the XMLs prettier, but I'm not sure how the other devs feel about that, since it's a potential cause for a very confusing issue in the future if we forget that.

Reply:
My gut tells me that it's better at the moment to remove the change in the library files, and accept the &apos; for the time being. We can come back to this in the future, but that would be my suggestion for having the true important part of this change (the XML spoiler log itself) included in the 2.0 release

Reply:
If the only issue with disabling entity processing is potential future issues with the & character, I think that's okay for now too. It seems very unlikely to come up in the future (for the most part all location and item names are pretty much set in stone) and if it does we can probably deal with it then imo